spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jms/activemq/PropertiesActiveMQConnectionDetails.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.jms.activemq;
+
+/**
+ * Adapts {@link ActiveMQProperties} to {@link ActiveMQConnectionDetails}.
+ *
+ * @author Eddú Meléndez
+ * @since 3.1.0
+ */
+public class PropertiesActiveMQConnectionDetails implements ActiveMQConnectionDetails {
```

Comment:
I think this could be package-private, and perhaps an inner-class of the `@Configuration` where it's defined.